hal/stm32l4.c

```diff
@@ -0,0 +1,270 @@
+/* stm32l4.c
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdint.h>
+#include <image.h>
+#include "stm32l4xx_hal.h"
+
+/* Assembly helpers */
+#define DMB() asm volatile ("dmb")
+
+/*** RCC ***/
+
+#define RCC_PRESCALER_DIV_NONE 0
+
+
+
+static uint32_t Address = 0, PAGEError = 0;
+
+
+static FLASH_EraseInitTypeDef EraseInitStruct;
+  
+  
+static uint32_t RAMFUNCTION GetPage(uint32_t Addr)
+{
+    uint32_t page = 0;
+  
+    if (Addr < (FLASH_BASE + FLASH_BANK_SIZE))
+        page = (Addr - FLASH_BASE) / FLASH_PAGE_SIZE;
+    else
+        page = (Addr - (FLASH_BASE + FLASH_BANK_SIZE)) / FLASH_PAGE_SIZE;
+  
+    return page;
+}
+
+
+static uint32_t RAMFUNCTION GetBank(uint32_t Addr)
+{
+    uint32_t bank = 0;
+  
+    if (READ_BIT(SYSCFG->MEMRMP, SYSCFG_MEMRMP_FB_MODE) == 0)
```

Comment:
For `if` and `else` blocks please put the brace at end like `if () {` and `else () {`. This conforms to existing coding styles. For functions we always put the brace on new line, so that is good.